src/main/kotlin/org/rust/lang/utils/evaluation/CfgEvaluator.kt

```diff
@@ -87,24 +86,29 @@ class CfgEvaluator(
         is CfgPredicate.Not -> !evaluatePredicate(predicate.single)
         is CfgPredicate.NameOption -> evaluateName(predicate.name)
         is CfgPredicate.NameValueOption -> evaluateNameValue(predicate.name, predicate.value)
-        is CfgPredicate.Feature -> evaluateFeature(predicate.name)
         is CfgPredicate.Error -> Unknown
     }
 
     private fun evaluateName(name: String): ThreeValuedLogic = when {
-        name in packageOptions.nameOptions -> True
-        // TODO: convert whitelist to blacklist and merge options with packageOption
-        name in SUPPORTED_NAME_OPTIONS -> ThreeValuedLogic.fromBoolean(options.isNameEnabled(name))
-        (name == "test" || name == "bootstrap" || name == "doc") && origin == PackageOrigin.STDLIB -> False
-        name == CfgOptions.TEST && isUnitTestMode -> ThreeValuedLogic.fromBoolean(options.isNameEnabled(name))
-        else -> Unknown
+        // https://doc.rust-lang.org/nightly/unstable-book/language-features/cfg-panic.html
+        name == "cfg_panic" -> Unknown
+
+        name == "test" && origin != PackageOrigin.STDLIB -> Unknown
+        strict -> ThreeValuedLogic.fromBoolean(options.isNameEnabled(name))
```

Comment:
Note that currently the logic is a bit duplicated, e.g. `ThreeValuedLogic.fromBoolean(options.isNameEnabled(name))` here and below in case of `name in SUPPORTED_NAME_OPTIONS`. I suppose strict evaluation can be performed in an easier way. AFAIU all we want to do is to replace `Unknown` with `False` in case of strict evaluation. All the evaluations like `options.isNameEnabled` or `options.isNameValueEnabled` are cheap, so we can perform them at first (like we did before), and then just replace `Unknown` with `False`.

What do you think about introducing such a method
```Kotlin
enum class ThreeValuedLogic {
    True, False, Unknown;

    // Not sure about the name, maybe `confirm` is better or even something else
    fun refine(strict: Boolean): ThreeValuedLogic = when (this) {
        True -> True
        False -> False
        Unknown -> if (strict) False else Unknown
    }
//...
}
```

and using it this way?
```Kotlin
private fun evaluateName(name: String): ThreeValuedLogic = when {
    //...
    name in SUPPORTED_NAME_OPTIONS -> ThreeValuedLogic.fromBoolean(options.isNameEnabled(name))
    else -> Unknown
}.refine(strict)
```

Reply:
I think this variant is much less explicit... and it saves only one line of code: `strict -> ThreeValuedLogic.fromBoolean(options.isNameEnabled(name))`. I don't this wirth it.